components/datadog/fakeintake/docker.go

```diff
@@ -0,0 +1,52 @@
+package fakeintake
+
+import (
+	"log"
+	"net"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/pulumi/pulumi-docker/sdk/v4/go/docker"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const hostPort = 30080
+
+func NewLocalDockerFakeintake(e config.CommonEnvironment, resourceName string) (*Fakeintake, error) {
+	return components.NewComponent(e, resourceName, func(comp *Fakeintake) error {
+
+		_, err := docker.NewContainer(e.Ctx, e.CommonNamer.ResourceName("local-docker-container"), &docker.ContainerArgs{
+			Image: pulumi.String("public.ecr.aws/datadog/fakeintake:latest"),
+			Ports: docker.ContainerPortArray{
+				&docker.ContainerPortArgs{
+					Internal: pulumi.Int(80),
+					External: pulumi.Int(hostPort),
+				},
+			},
+		}, e.WithProviders(config.ProviderDocker))
+		if err != nil {
+			return err
+		}
+
+		localIP := getLocalIP()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Ain't docker containers always listening on `localhost:<hostPort>`? You might skip this part

Reply:
Yes, if you want to reach the fakeintake from your computer you can do that. But I want a single URL that can be used from the your computer and from a `kind` (or `docker` image) running on your computer. Using `localhost:<hostPort>` in the `kind` cluster will not work. But using the host local IP does work.